### PR TITLE
Fix Email/Composer does not properly strip HTML body for message snippet preview 

### PR DESCRIPTION
**Pull request by yifanplanet**
# Code changes

- [x] Updated the HTML stripping function within composer component
- [x] Extract the stripping function into `utils.ts`
- [x] Remove the stripping from `hydrateDraftInThread` function in MailboxStore because we only need to generate snippet when user closes the composer.

# Readiness checklist

- [x] Cypress tests passing?
- [ ] New cypress tests added?

# License

I confirm that this contribution is made under the terms of the MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: components/composer/src/Composer.svelte [hunk 1]
```diff
@@ -162,6 +163,8 @@
     if (_this.reset_after_close) {
       resetAfterSend($message.from);
     }
```
Comment (yifanplanet) [inline components/composer/src/Composer.svelte:165]: This is the only place that snippet will be generated.

File: components/composer/src/lib/utils.ts [hunk 2]
```diff
@@ -16,6 +16,24 @@ export const debounce = (
...
+      // Remove style and script tag and content
+      .replace(/<style|script([\S\s]*?)>([\S\s]*?)<\/style|script>/g, "")
+      // Remove all tags and keep content / Remove all HTML entities
+      ?.replace(/<\/?[^>]+(>|$)|(&.+?;)/g, "")
+      // Remove extra spaces
+      ?.replace(/\s+/g, " ")
+      // Remove leading and trailing space
+      ?.trim()
+      // Add ... for longer message
+      ?.substring(0, 190) + (body.length > 190 ? "..." : "")
```
Comment (ozsivanov) [inline components/composer/src/lib/utils.ts:37]: Is this something that can be handled by the CSS instead? LGTM otherwise

**Review by ozsivanov — APPROVED** [review]

File: components/composer/src/lib/utils.ts [hunk 3]
```diff
@@ -16,6 +16,24 @@ export const debounce = (
     timeout = setTimeout(later, wait);
   };
 };
+
 export const isValidEmail = (email: string): boolean => {
   return /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email);
 };
+
+export const calculateMessageTempSnippet = (body: string): string => {
```
Comment (AaronDDM) [inline components/composer/src/lib/utils.ts:24]:
```suggestion
export const cleanMessageSnippet = (body: string): string => {
```

Not sure if we're doing a calculation here.

**Review by AaronDDM — APPROVED** [review]

File: components/composer/src/Composer.svelte [hunk 4]
```diff
@@ -162,6 +163,8 @@
...
+    // Update the snippet showing on the condensed draft message
+    msg.snippet = calculateMessageTempSnippet(msg.body);
```
Comment (AaronDDM) [inline components/composer/src/Composer.svelte:167]:
```suggestion
    msg.snippet = cleanMessageSnippet(msg.body);
```

File: components/composer/src/lib/utils.ts (hunk 2 above)
Comment (AaronDDM) [inline components/composer/src/lib/utils.ts:37]: @yifanplanet cold we instead use the same library we're using in the HTML body - `dompurify`? 
https://github.com/cure53/DOMPurify

Safer + easier to work with

File: components/composer/src/lib/utils.ts (hunk 3 above)
Comment (yifanplanet) [reply components/composer/src/lib/utils.ts:24]: @AaronDDM I basically just copied the similar name from the function they used for calculating the snippet in cloud-core. But yeah, cleanMessage makes more sense.

File: components/composer/src/lib/utils.ts (hunk 2 above)
Comment (yifanplanet) [reply components/composer/src/lib/utils.ts]: @AaronDDM Updated to code to use DOMPurify for a safer strategy agains xss.

Comment (AaronDDM) [reply components/composer/src/lib/utils.ts]: Amazing thanks @yifanplanet

**Pull request merged**